### PR TITLE
added params

### PR DESCRIPTION
**Pull request by AlexejKossmann**
This PR adds support for Parameters for QueryBuilder with IN and NOT IN operators

### CODE REVIEW

File: src/Database/DatabaseUtil.php [hunk 1]
```diff
@@ -622,15 +627,14 @@ public function composeWhereForQueryBuilder(QueryBuilder $queryBuilder, string $
...
-                                return '"'.addslashes(Controller::replaceInsertTags(trim($val), false)).'"';
-                            },
-                            $value
-                        )
+                    $where = $queryBuilder->expr()->in($field, $wildcard);
+                    $preparedValue = array_map(
+                        function ($val) {
+                            return addslashes(Controller::replaceInsertTags(trim($val), false));
+                        },
+                        $value
```
Comment (Defcon0) [inline src/Database/DatabaseUtil.php:635]: Why did you remove the double quotes? What if your field is a string field?

**Review by Defcon0 — CHANGES_REQUESTED** [review]

File: src/Database/DatabaseUtil.php [hunk 2]
```diff
@@ -566,6 +567,10 @@ public function composeWhereForQueryBuilder(QueryBuilder $queryBuilder, string $
             $wildcard = str_replace('.', '_', $wildcard);
```
Comment (koertho) [inline src/Database/DatabaseUtil.php:567]: WIldcard ist just the parameter name, if I read the code correcty? Wouldn't it be enought then, to check if a parameter name already exist and generate a new one by adding a suffix? So we would not need a manual additional parameter here.

Comment (Defcon0) [reply src/Database/DatabaseUtil.php:567]: Wrong line referenced? Did you mean line 570...572?

File: src/Database/DatabaseUtil.php (hunk 1 above)
Comment (AlexejKossmann) [reply src/Database/DatabaseUtil.php:635]: Double quotes are not possible, there happen some kind of shenenigans when doctrine gets an array with double quotes like [""2"",""3""] 
doctrine needs array for IN and NOT IN 
https://github.com/doctrine/dbal/blob/012569dfcc96478c061d3b529fecb5c8ac0405e5/src/Query/Expression/ExpressionBuilder.php#L281 

it should not be possible to add a string

File: src/Database/DatabaseUtil.php (hunk 2 above)
Comment (AlexejKossmann) [reply src/Database/DatabaseUtil.php:567]: what kind of prefix should this be? some random string? if so then this PR would not change anything, bcuz i want to address a specific Wildcard to change its value, e.g. in a EventListener

File: src/Database/DatabaseUtil.php (hunk 1 above)
Comment (AlexejKossmann) [reply src/Database/DatabaseUtil.php:635]: or we need convert a string into array before seting it as IN/NOT IN parameter

File: src/Database/DatabaseUtil.php (hunk 2 above)
Comment (koertho) [reply src/Database/DatabaseUtil.php:567]: @Defcon0 Shure :D (but since my comment was more generic, it wasn't so important)

@AlexejKossmann Therefore PRs have a description field to explain, why you do what you do ;)

What comes in mind: is it possible, that we need to add more parameters for this method in the future? If the answear !== no, then we should think about changing the parameter type to array with an id index, otherwise the parameter list will become even longer

Comment (Defcon0) [reply src/Database/DatabaseUtil.php:567]: @koertho is right. @AlexejKossmann Please add a parameter `array $options = []` and rename your `id` to `wildcardSuffix` and check it this way:

```
$wildcardSuffix = $options['wildcardSuffix'] ?? '';
```

OK for you both?

Comment (AlexejKossmann) [reply src/Database/DatabaseUtil.php:574]: @Defcon0 yes this is a nice solution! Adjusted the PR
